data_types/src/lib.rs

```
@@ -2111,14 +2111,12 @@ impl TimestampRange {
     ///
     /// Takes an inclusive start and an exclusive end. You may create an empty range by setting `start = end`.
     ///
-    /// Clamps start to [`MIN_NANO_TIME`].
-    /// end is unclamped - end may be set to i64:MAX == MAX_NANO_TIME+1 to indicate no restriction on time.
+    /// Clamps `start` to [`MIN_NANO_TIME`].
+    /// end is unclamped - end may be set to `i64:MAX == MAX_NANO_TIME+1` to indicate no restriction on time.
     ///
-    /// # Panic
-    /// Panics if `start > end`.
+    /// If `start > end`, this will be interpreted as an empty time range and `start` will be set to `end`.
     pub fn new(start: i64, end: i64) -> Self {
-        assert!(end >= start, "start ({start}) > end ({end})");
-        let start = start.max(MIN_NANO_TIME);
+        let start = start.max(MIN_NANO_TIME).min(end);
```

I worry that this will silently ignore errors that someone should be looking into. 

Should we add a `warn!` in this case? 

My rationale is that this case likely signals a bug in the client. But then maybe we should avoid spam in the iox log 🤔 

However, I don't have a single unified perfect suggestion, so I think this is fine too

---

Yeah, I would avoid spamming logs due to client imperfections.